src/transformStateWithClones.js

```diff
@@ -7,7 +7,89 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionTypes = [];
+  let REPEAT = false;
+
+  for (const action of actions) {
+    if (!actionTypes.includes(action.type)) {
+      actionTypes.push(action.type);
+    }
+  }
+
+  if (actionTypes.length === 1) {
+    REPEAT = true;
+  }
+
+  const currentState = { ...state };
```

Comment:
The variable name `currentState` should be more descriptive of its purpose as a copy of the original `state`. Consider renaming it to `stateCopy` to adhere to the naming conventions mentioned in the checklist.